src/planner/binder/expression/bind_function_expression.cpp

```diff
@@ -24,10 +24,26 @@ BindResult ExpressionBinder::BindExpression(FunctionExpression &function, idx_t
 	auto &catalog = Catalog::GetCatalog(context);
 	auto func = catalog.GetEntry(context, CatalogType::SCALAR_FUNCTION_ENTRY, function.schema, function.function_name,
 	                             false, error_context);
+
 	switch (func->type) {
 	case CatalogType::SCALAR_FUNCTION_ENTRY:
 		// scalar function
+
+		// scalar function with a lambda function
+		if (function.function_name == "list_transform" || function.function_name == "array_transform" ||
+		    function.function_name == "list_apply" || function.function_name == "array_apply") {
+			return BindLambdaFunction(function, (ScalarFunctionCatalogEntry *)func, depth, 1);
```

Comment:
Do we need to special-case this? This will make it impossible for e.g. extensions to define functions with lambas. Can we instead perhaps have a `LogicalType::LAMBDA` parameter (or similar) in the arguments that will trigger the `BindLambdaFunction` to be called?

Reply:
I replaced this with a loop to detect children of `expression_class == ExpressionClass::LAMBDA`. To make this more generic I need to check for the correct number of lambda parameters (exactly one for transform and filter) in their respective binds.
However, at that point the lambda expression is already a bound expression. I don't know how to best get the number of left-hand arguments from it, if I attempt to do this after the `IterateChildren` (will rename!) in the binder.